### PR TITLE
Add BackButton extension

### PR DESCRIPTION
**Pull request by arthuro555**
Adds the possibility to control the back button on mobile and browsers. Doesn't work on electron because electron doesn't have a back button.

### CODE REVIEW

**Comment by 4ian** [comment]
Thanks for the PR! 👍 I'll check the code. 
Should this be called BrowserBackButton? I'm afraid this could be confused with an extension to check if the Android "Back" button was pressed.

**Comment by arthuro555** [comment]
> to check if the Android "Back" button

It is for that as well

File: Extensions/BackButton.json [hunk 1]
```diff
@@ -0,0 +1,101 @@
+{
+  "author": "Arthur Pacaud (arthuro555)",
+  "description": "This extension prevents the back button from quitting the game and adds a condition to decide what it does instead.",
```
Comment (4ian) [inline Extensions/BackButton.json:3]: "Prevents the back button from quitting the game and add a condition to check when it's pressed (to allow customising its behavior)."

(I avoid saying "This extension", as it's implied when shown on screen)

File: Extensions/BackButton.json [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+          "parameterObjects": "",
+          "useStrict": true,
+          "eventsSheetExpanded": false
+        }
+      ],
+      "parameters": [],
+      "objectGroups": []
+    },
+    {
+      "description": "This simulate the normal action of the back button. This action will quit the app on mobile, and go back to the previous page on the web.",
```
Comment (4ian) [inline Extensions/BackButton.json:60]: "This simulate" => "This simulates"
"quit the app on mobile": "quit the app when in a mobile app"
"on the web": "when in a web browser"

**Review by 4ian — APPROVED** [review]
Looks great overall! Added a few nitpickings/typos to fix

**Comment by arthuro555** [comment]
I've fixed the wording issues you mentioned.

**Comment by 4ian** [comment]
Thanks! 👍

**Pull request merged**